collect_app/src/main/java/org/odk/collect/android/activities/viewmodels/CurrentProjectViewModel.kt

```diff
@@ -23,19 +26,36 @@ class CurrentProjectViewModel(
         currentProjectProvider.setCurrentProject(project.uuid)
         Analytics.log(AnalyticsEvents.SWITCH_PROJECT)
         analyticsInitializer.initialize()
-        refresh()
+        updateCurrentProjectIfNeeded()
     }
 
     fun refresh() {
+        updateCurrentProjectIfNeeded()
+
+        if (!File(storagePathProvider.getProjectRootDirPath()).exists()) {
+            Analytics.log(AnalyticsEvents.RECREATE_PROJECT_DIR)
+            storagePathProvider.getProjectDirPaths(currentProject.value.uuid).forEach { FileUtils.createDir(it) }
```

Comment:
This is duplicated code. I think instead of diving into addressing that, I can just work on a follow-up immediately